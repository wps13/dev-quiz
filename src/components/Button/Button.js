import React from 'react';
import {StyleSheet, TouchableOpacity, Text} from 'react-native';

const Button = ({text, pressFunction, buttonColor, textColor}) => {
  return (
    <TouchableOpacity
      onPress={() => pressFunction(text)}
      style={[styles.button, {backgroundColor: buttonColor}]}>
      <Text style={[styles.text, {color: textColor}]}>{text}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    shadowColor: '#2AC062',
    shadowOpacity: 0.4,
    shadowOffset: {height: 10, width: 0},
    shadowRadius: 20,
    padding: 10,
    borderRadius: 5,
    margin: 10,
    minWidth: 120,
    maxWidth: '85%',
  },
  text: {
    textAlign: 'center',
    fontSize: 16,
  },
});
