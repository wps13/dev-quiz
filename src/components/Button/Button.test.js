import React from 'react';
import {shallow} from 'enzyme';
import {Text} from 'react-native';

import Button from './Button';

const props = {
  text: 'Start',
  pressFunction: jest.fn(),
  buttonColor: '#333',
  textColor: '#FFF',
};

const renderButton = () => shallow(<Button {...props} />);

describe('Button', () => {
  it('should render correctly', () => {
    const wrapper = renderButton();
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should render the text correctly', () => {
    const wrapper = renderButton();

    const textComponent = wrapper.find('Text');

    expect(textComponent.exists()).toBe(true);
    expect(textComponent.dive().text()).toEqual(props.text);
    wrapper.unmount();
  });
  it('should call the press function correctly', () => {
    const wrapper = renderButton();
    wrapper.props().onPress();
    expect(props.pressFunction.mock.calls.length).toBe(1);
    wrapper.unmount();
  });
});
