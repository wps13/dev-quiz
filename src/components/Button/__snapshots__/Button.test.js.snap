// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "margin": 10,
        "maxWidth": "85%",
        "minWidth": 120,
        "padding": 10,
        "shadowColor": "#2AC062",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 20,
      },
      Object {
        "backgroundColor": "#333",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "textAlign": "center",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Start
  </Text>
</ForwardRef>
`;
